src/ImageData/StokesFilesConnector.cc

```diff
@@ -0,0 +1,324 @@
+    std::string prefix_file_name;                   // the common file name start from the first char
+    std::string postfix_file_name;                  // the common file name start from the last char
+    _concatenate_name = "hypercube_";               // initialize the name of concatenated image
+    ImageTypes image_types;                         // used to check whether the file type is the same
+
+    for (int i = 0; i < message.stokes_files_size(); ++i) {
+        auto stokes_file = message.stokes_files(i);
+        auto stokes_type = message.stokes_files(i).stokes_type();
+        casacore::String hdu(stokes_file.hdu());
+        casacore::String full_name(GetResolvedFilename(_top_level_folder, stokes_file.directory(), stokes_file.file()));
+
+        if (_loaders.count(stokes_type)) {
+            err = "Stokes type for is duplicate!\n";
+            return false;
+        }
+
+        // open an image file
+        if (!full_name.empty()) {
+            try {
+                if (hdu.empty()) { // use first when required
+                    hdu = "0";
+                }
+                _loaders[stokes_type].reset(carta::FileLoader::GetLoader(full_name));
+                _loaders[stokes_type]->OpenFile(hdu);
+            } catch (casacore::AipsError& ex) {
+                err = "Fail to open the file: " + ex.getMesg();
+                return false;
+            }
+        } else {
+            err = "File name is empty or does not exist!\n";
+            return false;
+        }
+
+        // update the concatenated image name and insert a new stokes type
+        _concatenate_name += CARTA::StokesType_Name(stokes_type);
+
+        // update the common file name starts from the head or tail
+        if (i == 0) {
+            image_types = CasacoreImageType(full_name);
+            prefix_file_name = stokes_file.file();
+            postfix_file_name = prefix_file_name;
+            std::reverse(postfix_file_name.begin(), postfix_file_name.end());
+        } else {
+            if (image_types != CasacoreImageType(full_name)) {
+                err = "Different file types can not be concatenated!\n";
+                return false;
+            }
+
+            // get the common file name start from the head
+            int tmp_pos_head = StrCmp(prefix_file_name, stokes_file.file());
+            if (tmp_pos_head < pos_head) {
+                pos_head = tmp_pos_head;
+            }
+            prefix_file_name = prefix_file_name.substr(0, pos_head);
+
+            // get the common file name start from the tail
+            std::string tmp_reverse_filename = stokes_file.file();
+            std::reverse(tmp_reverse_filename.begin(), tmp_reverse_filename.end());
+            int tmp_pos_tail = StrCmp(postfix_file_name, tmp_reverse_filename);
+            if (tmp_pos_tail < pos_tail) {
+                pos_tail = tmp_pos_tail;
+            }
+            postfix_file_name = postfix_file_name.substr(0, pos_tail);
```

Comment:
This can be done a lot more concisely with `std::mismatch`:

        auto common_prefix = [](std::string a, std::string b) {
                if(a.size() > b.size()) {
                        std::swap(a, b);
                }
                return std::string(a.begin(), std::mismatch(a.begin(), a.end(), b.begin()).first);
        };

What you are doing with the suffix (reversing, accumulating the prefix, reversing back at the end) is fine, and you can still do that using this lambda.

Reply:
I don't think we should use a custom function called `StrCmp`, as it will cause confusion with [`strcmp`](http://www.cplusplus.com/reference/cstring/strcmp/)